tests/python/test_fields_builder.py

```diff
@@ -3,174 +3,226 @@
-    fb2.dense(ti.i, n).place(z)
+    fb2.dense(ti.i, shape_size_1d).place(z)
     fb2.finalize()
 
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple():
+        for i in range(shape_size_1d):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape_size_1d):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape_size_1d):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_field_multiple()
+    for i in range(shape_size_1d):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape_size_1d):
         assert x[i] == i * 3
 
 
+'''
+Test fields with builder pointer.
+'''
+
+
 @ti.test(arch=[ti.cpu, ti.cuda, ti.metal])
 def test_fields_builder_pointer():
-    n = 5
-
+    shape_size_1d = 5
     fb1 = ti.FieldsBuilder()
     x = ti.field(ti.f32)
-    fb1.pointer(ti.i, n).place(x)
+    fb1.pointer(ti.i, shape_size_1d).place(x)
     fb1.finalize()
 
     @ti.kernel
-    def func1():
-        for i in range(n):
+    def assign_field_single():
+        for i in range(shape_size_1d):
             x[i] = i * 3
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape_size_1d):
         assert x[i] == i * 3
 
     fb2 = ti.FieldsBuilder()
     y = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(y)
+    fb2.pointer(ti.i, shape_size_1d).place(y)
     z = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(z)
+    fb2.pointer(ti.i, shape_size_1d).place(z)
     fb2.finalize()
 
-    # test range-for
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple0():
+        # test range-for
+        for i in range(shape_size_1d):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape_size_1d):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape_size_1d):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_field_multiple0()
+    for i in range(shape_size_1d):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    # test struct-for
     @ti.kernel
-    def func3():
+    def assign_field_multiple1():
+        # test struct-for
         for i in y:
             y[i] += 5
         for i in z:
             z[i] -= 5
 
-    func3()
-    for i in range(n):
+    assign_field_multiple1()
+    for i in range(shape_size_1d):
         assert y[i] == i + 10
         assert z[i] == i + 5
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape_size_1d):
         assert x[i] == i * 3
 
 
+'''
+Test fields with builder destory.
+'''
+
+
 @ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan])
 def test_fields_builder_destroy():
-    def A(i):
-        n = i * 10**3
+    test_sizes = [1]  # [1, 2, 3]
```

Comment:
We shouldn't have these comments in final version :-)

Reply:
fixed